.github/workflows/test.yml

```diff
@@ -982,122 +1060,179 @@ jobs:
-          echo "The following steps have failed in CI" > comment.md
-          echo >> comment.md
-
-      - name: Determine Jobs failed
-        if: always() && needs.determine_jobs.result != 'success' && needs.determine_jobs.result != 'skipped'
-        run: |
-          echo "- Determining jobs" >> failures.md
-          exit 1
-
-      - name: Go Linting failed
-        if: always() && needs.go_lint.result != 'success' && needs.go_lint.result != 'skipped'
-        run: |
-          echo "- Go lints" >> failures.md
-          exit 1
-
-      - name: Go Unit Tests failed
-        if: always() && needs.go_unit.result != 'success' && needs.go_unit.result != 'skipped'
-        run: |
-          echo "- Go Unit Tests" >> failures.md
-          exit 1
-
-      - name: Go Integration Tests failed
-        if: always() && needs.go_integration.result != 'success' && needs.go_integration.result != 'skipped'
-        run: |
-          echo "- Go Integration Tests" >> failures.md
-          exit 1
-
-      - name: Go Examples failed
-        if: always() && needs.go_examples.result != 'success' && needs.go_examples.result != 'skipped'
-        run: |
-          echo "- Go Examples" >> failures.md
-          exit 1
-
-      - name: Go E2E failed
-        if: always() && needs.go_e2e.result != 'success' && needs.go_e2e.result != 'skipped'
-        run: |
-          echo "- Go E2E" >> failures.md
-          exit 1
-
-      - name: Rust Compilation failed
-        if: always() && needs.rust_prepare.result != 'success' && needs.rust_prepare.result != 'skipped'
-        run: |
-          echo "- Rust compilation" >> failures.md
-          exit 1
-
-      - name: Rust Linting failed
-        if: always() && needs.rust_lint.result != 'success' && needs.rust_lint.result != 'skipped'
+      - name: Compute info
+        id: info
+        if: always()
         run: |
-          echo "- Rust lints" >> failures.md
-          exit 1
+          cancelled=false
+          failure=false
+          subjob () {
+            local result=$1
+            local name=$2
+            if [ "$result" = "cancelled" ]; then
+              cancelled=true
+            elif [ "$result" != "success" ] && [ "$result" != "skipped" ]; then
+              echo "- $name" >> failures.md
+              failure=true
+            fi
+          }
+          subjob ${{needs.determine_jobs.result}} "Determining jobs"
+          subjob ${{needs.go_lint.result}} "Go lints"
+          subjob ${{needs.go_unit.result}} "Go unit tests"
+          subjob ${{needs.go_examples.result}} "Go examples"
+          subjob ${{needs.go_e2e.result}} "Go e2e tests"
+          subjob ${{needs.go_integration.result}} "Go integration tests"
+          subjob ${{needs.rust_prepare.result}} "Rust prepare"
+          subjob ${{needs.rust_lint.result}} "Rust lints"
+          subjob ${{needs.rust_check.result}} "Rust checks"
+          subjob ${{needs.turbopack_rust_test1.result}} "Turbopack Rust tests (linux)"
```

Comment:
It's weird that this name is different from that of the job itself (Turbopack Rust testing on ubuntu).

Reply:
We might move more tests from 2 to 1 when they have faster runners... I think it's more part 1 and 2, instead of linux in particular

Reply:
Maybe rename to (blocking) and (non-blocking)?